src/components/productrecommendation.fetchallpages.jsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright © 2018 Elastic Path Software Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { login } from '../utils/AuthService';
+import ProductListItemMain from './productlistitem.main';
+
+const Config = require('Config');
+
+
+class Productrecommendationfetchallpages extends React.Component {
+  static propTypes = {
+    product: PropTypes.objectOf(PropTypes.any).isRequired,
+    length: PropTypes.number.isRequired,
+    count: PropTypes.number.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      links: [],
+    };
+  }
+
+  componentDidMount() {
+    const { product } = this.props;
+    const { links } = this.state;
+    const link = links;
+    const arr = product.links;
+    for (let i = 0; i < arr.length; i += 1) {
+      if (arr[i].rel === 'element') {
+        link.push(arr[i]);
+      } else if (arr[i].rel === 'next') {
+        this.renderUrl(arr[i].uri);
+      }
+    }
+    this.setState({
+      links: link,
+    });
+  }
+
+  renderUrl(uri) {
+    const { links } = this.state;
+    const link = links;
+    login().then(() => {
+      fetch(`${Config.cortexApi.path + uri}`,
+        {
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+          },
+        })
+        .then(res => res.json())
+        .then((res) => {
+          const arr = res.links;
+          for (let i = 0; i < arr.length; i += 1) {
+            if (arr[i].rel === 'element') {
+              link.push(arr[i]);
+            } else if (arr[i].rel === 'next') {
+              this.renderUrl(arr[i].uri);
+            }
+          }
+          this.setState({
+            links: link,
+          });
+          return res;
+        });
+    });
+  }
+
+  static renderProducts(product, length, MaxItemsInOneCarouselView) {
```

Comment:
you don't use length, why is it here?

Reply:
hmm i am supposed to. But due to cortex bug.. the length value is incorrect. I would rather keep it there, so when the bug is fixed, its an easy fix to redirect to original length

Reply:
hmm, you actually shouldn't need to use length though, you should be able to do things inside a for..in loop, or using map or something instead.